common/djangoapps/lang_pref/middleware.py

```diff
@@ -26,3 +28,14 @@ def process_request(self, request):
             # Set it to the LANGUAGE_SESSION_KEY (Django-specific session setting governing language pref)
             if user_pref:
                 request.session[LANGUAGE_SESSION_KEY] = user_pref
+            else:
+                # Set browser language if it is supported by system.
+                preferred_language = request.META.get('HTTP_ACCEPT_LANGUAGE', '')
+                lang_headers = [seq[0] for seq in parse_accept_lang_header(preferred_language)]
+                if lang_headers:
+                    # lang_header is list of tuple
+                    # language with highest priority in browser will be at zero index.
+                    browser_lang = lang_headers[0]
```

Comment:
I think this logic should loop through all of the browser's language preferences, and match the first one that is available. For example, a user might prefer Swiss German, then regular German, and then finally English. This logic would fail to match German even if we have translations, because we don't have Swiss German.